contents/docs/sdk/features/party/index.mdx

```diff
@@ -7,24 +7,94 @@ description: How to use the Mineplex Studio Party Module.
+
+Fetching a party can be done using its ID via `getPartyById(String)` or by a given player using `getPartyOfPlayer(Player)`. 
+
+```java
+final PartyModule partyModule = MineplexModuleManager.getRegisteredModule(PartyModule.class);
+if (partyModule.isLoaded(player)) {
+    // Safe to check if they have a party.
+    final Optional<Party> party = partyModule.getPartyOfPlayer(player);
+}
+```
+
+## Retrieving Party Details
+
+Once you've retrieved a party, you can access its ID, its leader, and all players in the party.
+
+### Party ID
+
+All parties have a unique ID assigned that can be retrieved via `getId()`.
+
+```java
+final String partyId = party.getId();
+```
+
+### Party Members
+
+<Note type="warning">
+    There is no guarantee that players in the party will be online. Parties persist until disbanded.
+</Note>
+
+#### Retrieving Party Leader 
+
+Given a party, you can call `getOwner()` to return a `OfflinePlayer` of the owner of the party.
+
+```java
+final OfflinePlayer partyLeader = party.getOwner();
+if (partyLeader.isOnline()) {
+    Bukkit.broadcast(Component.text("Party leader is in-game!"));
+} else {
+    Bukkit.broadcast(Component.text("Party leader is not in this game!"));
+}
+```
+
+#### Retrieving Party Members
+
+Party members can be retrieved from a party by calling `listMembers(boolean)` with the argument being whether to include the party owner in the returned collection of `OfflinePlayer` objects.
+
+```java
+final OfflinePlayer partyOwner = party.getOwner();
+
+// We specified false, meaning that this will NOT include the party leader.
+for (final OfflinePlayer offlinePlayer : party.listMembers(false)) {
+    if (offlinePlayer.isOnline()) {
+        final Player player = (Player) offlinePlayer;
+        player.sendMessage("You are a party member of " + partyOwner.getName() + "!");
+    }
+}
+```
+
+#### Checking Party Members
+
+You can also verify that a player is a member of a party via `hasMember(OfflinePlayer)` or `hasMember(UUID)`, where the UUID is the UUID of the player you are checking.
+
+```java
+final boolean isPartyMember = party.hasMember(player);
+if (isPartyMember) {
+    player.sendMessage("You are a member of this party!");
+}
+```
+
+
+## Example
+Let's say we are making an open-world game where players can use the party system to team up. We don't want to allow players in the same party to hurt one another, so we integrate with the party module.
+```java
+public boolean canDamage(final Player damager, final Player victim) {
+    final Optional<Party> damagerParty = partyModule.getPartyOfPlayer(damager);
+    final Optional<Party> victimParty = partyModule.getPartyOfPlayer(victim);
+
+    // If neither player is in a party, they can't be in the same party
+    if (damagerParty.isEmpty() || victimParty.isEmpty()) {
+        return true;
+    }
+
+    // If the players are in one party, they can't hurt one another
+    return !damagerParty.get().hasMember(victim);
+}
+```
```

Comment:
I'd generally say this is a bit too complex.

```java
public boolean canDamage(final Player damager, final Player victim) {
    return partyModule.getPartyOfPlayer(damager)
            .flatMap(damagerParty -> 
                partyModule.getPartyOfPlayer(victim)
                           // if the victim has a party - return false if it's the same party (cannot damage) or true otherwise.
                           .map(victimParty -> !damagerParty.equals(victimParty)))
            // if they aren't in a party - they can damage
            .orElse(true);
}
```

Unsure if newer developers would understand this though

Reply:
I'd say that the new example you have there would be more difficult to understand/more complex compared to the existing example. It's less straight forward and I'd say that most people typically wouldn't know about `flatMap` and would have a harder time trying to understand that compared to their existing knowledge of `Optional`s (since the SDK already uses that elsewhere)

Reply:
do you have anything in mind to remove that .get() call, I'm a bit worried people would try and modify the example and get a bit tripped up on that. That's why I presented that.

Reply:
What if we just replaced the last return stmt with 
```java
// If the attacker's party has the victim, we cannot attack them.
boolean damagerPartyContainsVictim = damagerParty.map(party -> party.hasMember(victim)).orElse(false);
return !damagerPartyContainsVictim;
```
? It still utilizes `map` unfortunately, but it's less wrapping than the `flatMap` example.

Alternatively maybe a comment saying "Make sure to check that the party exists first!" ?

Reply:
We could also just get rid of the example entirely really. The code snippets above it are basically examples.

Reply:
minus the `final` on boolean, seems good to me

Reply:
Ended up removing the `if` stmt as well since we're using `map` and `orElse` anyway.